File: bindings/java/src/main/java/com/xfa/pdf/PdfException.java
package com.xfa.pdf;

/**
 * Exception thrown by PDF operations when the native engine encounters an error.
 */
public class PdfException extends RuntimeException {
    public PdfException(String message) {
        super(message);
    }

    public PdfException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: bindings/dotnet/src/XfaPdf/PdfException.cs
using System;

namespace XfaPdf
{
    /// <summary>
    /// Exception thrown when a PDF operation fails.
    /// </summary>
    public class PdfException : Exception
    {
        /// <summary>
        /// The native status code that caused this exception.
        /// </summary>
        public PdfStatus Status { get; }

        public PdfException(PdfStatus status, string message)
            : base(message)
        {
            Status = status;
        }

        public PdfException(string message)
            : base(message)
        {
            Status = PdfStatus.ErrorUnknown;
        }

        public PdfException(string message, Exception innerException)
            : base(message, innerException)
        {
            Status = PdfStatus.ErrorUnknown;
        }
    }
}
